pkg/registry/apps/statefulset/strategy.go

```diff
@@ -112,7 +113,9 @@ func (statefulSetStrategy) AllowCreateOnUpdate() bool {
 func (statefulSetStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object) field.ErrorList {
 	newStatefulSet := obj.(*apps.StatefulSet)
 	oldStatefulSet := old.(*apps.StatefulSet)
-	validationErrorList := validation.ValidateStatefulSet(newStatefulSet)
+
+	opts := pod.GetValidationOptionsFromPodTemplate(&newStatefulSet.Spec.Template, &newStatefulSet.Spec.Template)
```

Comment:
just noticed this is passing newStatefulSet as both args

Reply:
Changed. a unit test for this one is going to look strange because statefulset is not supposed to use this.